src/main/java/net/oauth/jsontoken/JsonTokenParser.java

```diff
@@ -257,11 +253,11 @@ public boolean issuedAtIsValid(JsonToken jsonToken, Instant now) {
    * @throws IllegalStateException if tokenString is not a properly formatted JWT
    */
   private String[] splitTokenString(String tokenString) {
-    String[] pieces = tokenString.split(Pattern.quote(JsonTokenUtil.DELIMITER));
-    if (pieces.length != 3) {
+    List<String> pieces = Splitter.on(JsonTokenUtil.DELIMITER).limit(4).splitToList(tokenString);
+    if (pieces.size() != 3) {
       throw new IllegalStateException("Expected JWT to have 3 segments separated by '" + 
-          JsonTokenUtil.DELIMITER + "', but it has " + pieces.length + " segments");
+          JsonTokenUtil.DELIMITER + "', but it has " + pieces.size() + " segments");
     }
-    return pieces;
+    return pieces.toArray(new String[0]);
```

Comment:
nit: Can we return and use the List<String> instead of converting it to an array?

Reply:
Changed here and the rest of the class